docs/dev.rst

```diff
@@ -1,20 +1,540 @@
+
+To run the ``update_dependencies`` task: 
+
+.. code-block:: console
+
+    $ cci task run update_dependencies
+
+
+Managed Package Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Managed package dependencies are rather simple. Under the ``project__dependencies`` section of the ``cumulusci.yml`` file, specify the namespace of the target package, and the required version number.
+
+    Example: ``npe01 version 3.6``
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - namespace: npe01
+              version: 3.6
+
+
+
+Automatic Install, Upgrade, or Uninstall/Install
+************************************************
+
+When the ``update_dependencies`` task runs, it retrieves a list of all managed packages in the target org, and creates a list of the installed packages and their version numbers.
+
+    Example: ``npe01 version 3.6``
+    
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - namespace: npe01
+              version: 3.6
+    
+..    
+    
+    Depending on whether or not the package with namespace ``npe01`` is installed, the ``update_dependencies`` task runs these steps. 
+
+    * If ``npe01`` is not installed, ``npe01 version 3.6`` is installed.
+    * If the org already has ``npe01 version 3.6`` installed, no changes take place.
+    * If the org has an older version installed, it's upgraded to ``version 3.6``.
+    * If the org has a newer version or a beta version installed, it's uninstalled and ``version 3.6`` is installed.
+
+
+Hierarchical Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Managed package dependencies can handle a hierarchy of dependencies between packages.
+
+    Example: Salesforce.org's Nonprofit Success Pack (NPSP), an extension of five other managed packages, one of which (Households) is an extension of another (Contacts & Organizations).
+
+    These dependencies are listed under the ``project`` section of the ``cumulusci.yml`` file.
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - namespace: npo02
+              version: 3.8
+              dependencies:
+                  - namespace: npe01
+                    version: 3.6
+            - namespace: npe03
+              version: 3.9
+            - namespace: npe4
+              version: 3.5
+            - namespace: npe5
+              version: 3.5
+
+..
+
+    The project requires ``npo02 version 3.8``, which itself requires ``npe01 version 3.6``. By specifying the dependency hierarchy, the ``update_dependencies`` task is capable of uninstalling and upgrading packages intelligently.
+
+    So if the target org currently has ``npe01 version 3.7``, ``npe01`` needs to be uninstalled to downgrade to ``3.6``. However, ``npo02`` requires ``npe01``, so uninstalling ``npe01`` also requires uninstalling ``npo02``. (In this scenario ``npe03``, ``npe04``, and ``npe05`` do not have to be uninstalled to uninstall ``npe01``.)
```

Comment:
@Br4nd0R @davidmreed I changed the previous example (npo02/npe01) to the new one (Households/Contacts & Organizations), but now it doesn't match the rest of the commentary and code block? How do we go about fixing this? Update the information to match Household/C&O, or go back to what we had previously? 

(FYI, it was previously:

 "Example: Salesforce.org's Nonprofit Success Pack (NPSP), an extension of five other managed packages, one of which (npo02) is an extension of another (npe01).")

Reply:
can we add the namespace along with the product name above? This way users will be able to associate between product <--> namespace.